helper/include/odp/helper/cli.h

```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2021, Nokia
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+/**
+ * @file
+ *
+ * ODP CLI helper API
+ *
+ * This API allows control of ODP CLI server, which may be connected to
+ * using a telnet client. CLI commands may be used to get information
+ * from an ODP instance, for debugging purposes.
+ *
+ * Many CLI commands output the information to the console, or wherever
+ * ODP logs have been directed to in global init.
+ */
+
+#ifndef ODPH_CLI_H_
+#define ODPH_CLI_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp_api.h>
+#include <odp/helper/ip.h>
+#include <stdint.h>
+
+/**
+ * @addtogroup odph_cli ODPH CLI
+ * @{
+ */
+
+/** ODP CLI server parameters */
+typedef struct {
+	/**
+	 * A character string containing an IP address. Default is
+	 * "127.0.0.1".
+	 */
+	char address[ODPH_IPV6ADDRSTR_LEN];
```

Comment:
This could be "const char *address" instead of table. It would be easier for application to give pointer to string than copy it. Helper needs to copy the address anyway (if it needs to save it for later use). This API spec can require that max string length is ODPH_IPV6ADDRSTR_LEN for IPv6 and ODPH_IPV4ADDRSTR_LEN for IPv4.

Reply:
I thought it might be a bad idea to return a pointer to string in odph_cli_param_init(), but okay. At least currently cli helper doesn't need a copy of the string, so with the parameter changed to a pointer, there's no need for any length information anywhere, and no need to introduce the ODPH_IPV?ADDRSTR_LEN defines either. So I'm dropping that commit.